src/rules/seczetta-grab-risk-score.js

```diff
@@ -0,0 +1,223 @@
+          "object_type": "NeAttribute",
+          "comparison_operator": "==",
+          "value": uid
+        }
+      ]
+    }
+  };
+  
+  try {
+    	profileResponse = await axios.post(profileRequestUrl,advancedSearchBody,{
+        headers: {
+          'Content-Type':'application/json',
+          'Authorization': 'Token token='+configuration.SECZETTA_API_KEY,
+          'Accept': 'application/json'
+        },
+      });
+
+      //if the user isnt found via the advanced search. A 
+    	if( profileResponse.data.profiles.length === 0 ) {
+        console.log("Profile not found. Empty Array sent back!");
+  	 		if( configuration.SECZETTA_AUTHENTICATE_ON_ERROR && configuration.SECZETTA_AUTHENTICATE_ON_ERROR === "true" ) {
+          return callback(null, user, context);
+        }
+        return callback(new UnauthorizedError("Error retrieving Risk Score."));
+  		}
+    
+  } catch (profileError) {
+    // Swallow risk scope API call, default is set to highest risk below.
+    console.log(`Error while calling Profile API: ${profileError.message}`);
+    if( configuration.SECZETTA_AUTHENTICATE_ON_ERROR && configuration.SECZETTA_AUTHENTICATE_ON_ERROR === "true" ) {
+      return callback(null, user, context);
+    }
+    return callback(new UnauthorizedError("Error retrieving Risk Score."));
+  }
+
+  //Should now have the profile in profileResponse. Lets grab it.
+  let objectId = profileResponse.data.profiles[0].id;
+  console.log(objectId);
+
+  let riskScoreRequestUrl = configuration.SECZETTA_BASE_URL + '/risk_scores?object_id=' + objectId;
+
+  try {
+      riskScoreResponse = await axios.get(riskScoreRequestUrl,{
+      headers: {
+        'Content-Type':'application/json',
+        'Authorization': 'Token token='+configuration.SECZETTA_API_KEY,
+        'Accept': 'application/json'
+      },
+    });
+  } catch (riskError) {
+    // Swallow risk scope API call, default is set to highest risk below.
+    console.log(`Error while calling Risk Score API: ${riskError.message}`);
+    if( configuration.SECZETTA_AUTHENTICATE_ON_ERROR && configuration.SECZETTA_AUTHENTICATE_ON_ERROR === "true" ) {
+      return callback(null, user, context);
+    }
+    return callback(new UnauthorizedError("Error retrieving Risk Score."));
+  }
+
+  //Should now finally have the risk score. Lets add it to the user
+  var riskScoreObj = riskScoreResponse.data.risk_scores[0];
+  var overallScore = riskScoreObj.overall_score;
+
+  // Default risk value is set to highest if API fails or no score returned.
+  //var riskScore = typeof apiResponse.riskScore === "number" ? riskScore : 100;
+
+  const allowableRisk = parseInt(configuration.SECZETTA_ALLOWABLE_RISK, 10);
+  const maximumRisk = parseInt(configuration.SECZETTA_MAXIMUM_ALLOWED_RISK, 10);
+  
+  //if risk score is below the maxium risk score but above allowable risk: Require MFA
+  if (allowableRisk && overallScore > allowableRisk && overallScore < maximumRisk) {
```

Comment:
```suggestion
  if (allowableRisk && overallScore => allowableRisk && overallScore <= maximumRisk) {
```

Seems like this is a bit more accurate but I could be wrong.

Reply:
Also, what if `allowableRisk === 0`? Is that something that a customer might do intentionally in a risk where maybe their risk profile is temporarily higher?

Reply:
I think if allowableRisk is 0 then it should be fine. MFA would prompt for any user with a risk score 0.1 or above (less than Max). The only way someone would get thru without MFA is if they didnt have a risk score at all or a risk score of 0. And if they really just want to do MFA for an app they likely wouldnt be using this rule.

I dont think we need the the `=>` or `<=` either because if 7.0 is the allowable risk score and the user has a 7.0 risk score we would want them to continue successfully without prompting. I believe i am thinking about this correctly :)

Reply:
RE: `allowableRisk` being zero ...  this conditional is looking for "existence" of `allowableRisk` so if it was set to zero then this branch would be skipped. I'm thinking if someone set that to `0` if, say, they were getting attacked or needed higher security for a short time? Edge case but easy to address.

RE: inequality operators ... you're totally right. Programming is hard!